freeipa/src/main/java/com/sequenceiq/freeipa/service/config/RootCertRegisterService.java

```diff
@@ -68,6 +63,34 @@ public RootCert register(Stack stack) throws FreeIpaClientException {
         return registerInternal(stack);
     }
 
+    @Override
+    public void delete(Stack stack) {
+        try {
+            rootCertService.deleteByStack(stack);
+        } catch (NotFoundException e) {
+            LOGGER.info("Root cert not exists for environment {}", stack.getEnvironmentCrn());
+        }
+    }
+
+    private RootCert registerInternal(Stack stack) throws FreeIpaClientException {
+        MDCBuilder.buildMdcContext(stack);
+        String rootCertificate = getRootCertFromFreeIpa(stack);
+        Optional<RootCert> currentCert = rootCertService.findByStackId(stack.getId());
+        return currentCert.isPresent() ? updateRootCertEntityIfNecessary(currentCert.get(), rootCertificate) : createRootCertEntity(stack, rootCertificate);
+    }
+
+    private RootCert updateRootCertEntityIfNecessary(RootCert rootCert, String rootCertificate) {
+        String rootCertInPemFormat = convertToPemFormat(rootCertificate);
+        if (!rootCert.getCert().equals(rootCertInPemFormat)) {
+            rootCert.setCert(rootCertInPemFormat);
+            LOGGER.debug("Updating root cert entity with new certificate.");
+            return rootCertService.save(rootCert);
+        } else {
+            LOGGER.debug("The root cert entity is already present. Not necessary to update.");
+        }
+        return rootCert;
```

Comment:
nit: this could move into the `else` branch so it would be obvious that this and the log message belong together